### PR TITLE
Decouple SyntaxList from SyntaxListBuilder.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This will make it easier to move SyntaxList to the core layer.

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-compiler

File: src/Compilers/CSharp/Portable/Parser/Lexer.cs [hunk 1]
```diff
@@ -303,22 +303,24 @@ internal SyntaxTriviaList LexSyntaxLeadingTrivia()
         {
             _leadingTriviaCache.Clear();
             this.LexSyntaxTrivia(afterFirstToken: TextWindow.Position > 0, isTrailing: false, triviaList: ref _leadingTriviaCache);
-            return new SyntaxTriviaList(default(Microsoft.CodeAnalysis.SyntaxToken), SyntaxList.List(_leadingTriviaCache), 0, 0);
+            return new SyntaxTriviaList(default(Microsoft.CodeAnalysis.SyntaxToken),
+                _leadingTriviaCache?.ToListNode(), position: 0, index: 0);
```
Comment (cston) [inline src/Compilers/CSharp/Portable/Parser/Lexer.cs]: It looks like `_leadingTriviaCache` and `_trailingTriviaCache` are never null.

File: src/Dependencies/PooledObjects/Microsoft.CodeAnalysis.PooledObjects.shproj [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <PropertyGroup Label="Globals">
-    <ProjectGuid>cd77a8cc-2885-47a7-b99c-fbf13353400b</ProjectGuid>
+    <ProjectGuid>c1930979-c824-496b-a630-70f5369a636f</ProjectGuid>
```
Comment (cston) [inline src/Dependencies/PooledObjects/Microsoft.CodeAnalysis.PooledObjects.shproj]: Change can be reverted.

**Comment by cston** [comment]
LGTM

File: src/Compilers/CSharp/Portable/Parser/Lexer.cs (hunk 1 above)
Comment (CyrusNajmabadi) [inline src/Compilers/CSharp/Portable/Parser/Lexer.cs]: Addressed.

**Comment by CyrusNajmabadi** [comment]
ping any other @dotnet/roslyn-compiler  devs?

**Comment by gafter** [comment]
LGTM

**Comment by VSadov** [comment]
LGTM

**Pull request merged**